tests/integration/multinodes.nim

```diff
@@ -103,9 +103,9 @@ template multinodesuite*(name: string,
         "prover",
         "--eth-account=" & $accounts[index],
         # TODO: this is wrong, client shouldn't need circom stuff
-        "--circom-r1cs=tests/circuits/fixtures/proof_main.r1cs",
-        "--circom-wasm=tests/circuits/fixtures/proof_main.wasm",
-        "--circom-no-zkey"]
+        "--circom-r1cs=vendor/codex-contracts-eth/verifier/networks/hardhat/proof_main.r1cs",
```

Comment:
I think it's probably better to store codex specific assents in codex itself, preferably in `tests/circuits/fixtures`? It's probably fine to duplicate them for the contracts, but for codex tests, I'd rather we keep them in the same codebase.

Reply:
The problem that I tried to solve is that the circuit assets need to match between the codex-contracts-eth and nim-codex. And since nim-codex already has the contracts in its vendor folder, it seemed easiest to put the asset with the contracts.

You could also copy them into the codex-contracts-eth from nim-codex during testing, but that seemed less elegant?

Reply:
Well, the point is that in the context of codex, and specifically the integration tests, we're trying to make sure that all the subsystems - storage/networking, block selection and sampling as well as the contracts and marketplace, function correctly. 

There are a set of assets that are used in codex proper, regardless of what tests the individual components rely on and this should be under codex, because as I already said, they are shared by several subsystems. In this case, the integration tests aren't testing _just_ the contracts, they rely on all this subsystems functioning for the contract to execute correctly and we better test all of this against the same set of assets which, IMO, should be in the codex repo. Then each project, external to codex, can contain it's own set of assets as well, but this are unrelated to those under codex.

Reply:
That's fine too :+1:  I have no strong opinion on this.